backend/server.py

```diff
@@ -141,20 +139,35 @@ def delete_source(source_id: int, _: dict = Depends(get_user_info)):
     return rows_deleted
 
 ####### SEGMENTS 
-@app.get('/segments/{source_id}', response_model=list[dict])
-def read_sources(source_id: int, user_info: dict = Depends(get_user_info)):
+@app.get('/segments/{source_id}', response_model=dict)
+def read_segments(source_id: int, offset: int = 0, limit: int = 100, user_info: dict = Depends(get_user_info)):
```

Comment:
I don't see where you select the latest segment by timestamp.

IIRC each segment can be saved several times and we need to fetch the latest, that is what the get_latest_segments should do.